packages/electron/src/electron/main/index.ts

```diff
@@ -2,10 +2,74 @@ import path from "node:path";
 import * as Sentry from "@sentry/electron/main";
 import { clearBuffer, setupSidecar } from "@spotlightjs/sidecar";
 import { BrowserWindow, Menu, app, dialog, ipcMain, shell } from "electron";
+import { autoUpdater as nativeUpdater } from "electron";
 import Store from "electron-store";
+import { type UpdateCheckResult, autoUpdater } from "electron-updater";
 
 const store = new Store();
 
+autoUpdater.forceDevUpdateConfig = true;
+
+const ONE_HOUR = 60 * 60 * 1000;
+let updateInfo: UpdateCheckResult | null;
+
+function installAndRestart() {
+  /**
+   * On macOS 15+ auto-update / relaunch issues:
+   * - https://github.com/electron-userland/electron-builder/issues/8795
+   * - https://github.com/electron-userland/electron-builder/issues/8997
+   */
+  if (process.platform === "darwin") {
+    app.removeAllListeners("before-quit");
+    app.removeAllListeners("window-all-closed");
+
+    for (const win of BrowserWindow.getAllWindows()) {
+      if (win.isDestroyed()) continue;
+      win.removeAllListeners("close");
+      win.close();
+    }
+
+    nativeUpdater.once("before-quit-for-update", () => {
+      app.exit();
+    });
+  }
+
+  autoUpdater.quitAndInstall();
+}
+
+async function checkForUpdates() {
+  let timeout = 24 * ONE_HOUR;
+  try {
+    updateInfo = await autoUpdater.checkForUpdates();
+
+    if (!updateInfo?.isUpdateAvailable) {
+      timeout = 3 * ONE_HOUR;
+    }
+  } catch (error) {
+    console.error(error);
+  } finally {
+    setTimeout(checkForUpdates, timeout);
+  }
+}
+
+app.on("ready", () => {
+  checkForUpdates();
+
+  autoUpdater.on("update-downloaded", () => {
+    dialog
+      .showMessageBox({
+        type: "question",
+        message: "A new update has been downloaded. It will be installed on restart.",
+        buttons: ["Restart", "Later"],
+      })
+      .then(result => {
+        if (result.response === 0) {
+          installAndRestart();
+        }
+      });
+  });
```

Comment:
Can we not make the handler async and writh this as
```suggestion
  autoUpdater.on("update-downloaded", async () => {
    const {response} = await dialog
      .showMessageBox({
        type: "question",
        message: "A new update has been downloaded. It will be installed on restart.",
        buttons: ["Restart", "Later"],
      });
      if (response === 0) {
        installAndRestart();
      }
  });
```